tokio/src/sync/semaphore.rs

```diff
@@ -73,6 +73,21 @@ use std::sync::Arc;
 /// }
 /// ```
 ///
+/// Limit access to a file:
+/// ```
+/// use std::io::{Result, Write};
+/// use std::fs::File;
+///
+/// static PERMITS: Semaphore = Semaphore::const_new(100);
+///
+/// async fn write_to_file(message: &[u8]) -> Result<()> {
+///     let _permit = PERMITS.acquire().await.unwrap();
+///     let mut buffer = File::create("example.txt")?;
+///     buffer.write_all(message)?;
+///     Ok(())
+/// }
+/// ```
```

Comment:
It seems like the test doesn't currently compile. You can try it yourself with `cargo test --doc` to get the compilation errors. You can also see them in the list of CI checks by opening more details on the failed tests.

Reply:
Fixed!